service/verify.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/slsa-framework/slsa-verifier/cmd"
+	"github.com/slsa-framework/slsa-verifier/verification"
+)
+
+var errInvalid = errors.New("invalid")
+
+type v1Query struct {
+	// Compulsory fields.
+	Source       string `json:"source"`
+	ArtifactHash string `json:"artifactHash"`
+	DsseEnvelope string `json:"provenanceContent"`
+	// Optional fields.
+	Tag             *string `json:"tag"`
+	Branch          *string `json:"branch"`
+	VersionedTag    *string `json:"versionedTag"`
+	PrintProvenance *bool   `json:"printProvenance"`
+}
+
+type validation string
+
+var (
+	validationSuccess = validation("success")
+	validationFailure = validation("failure")
+)
+
+type v1Result struct {
+	Version         uint       `json:"version"`
+	Error           *string    `json:"error,omitempty"`
+	Validation      validation `json:"validation"`
```

Comment:
Maybe an exit status int? easier than a string for version compatibility in case we extend this later.

Reply:
string can also be extended, right?
I'm not against using code, just trying to see the additional benefits it gives us.

I'm thinking we may want an `ErrorMsg`, `ErrorCode`... but I'm not sure if Golang handles that easily in practice...unless we wrap all errors with our own structure. Wdut?